### PR TITLE
Expose bulk query functionality for the Bulk API

### PR DESCRIPTION
**Pull request by k4jiang**
Reference: https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_using_bulk_query.htm

Expose functionality to perform Bulk Queries and fetch for batch results when dealing with larger datasets.
This includes two new methods:
1) Get list of Batch Result IDs
ex) `GET baseURI/job/750x00000000014/batch/751x00000000030/result`
2) Get Bulk Query Results
ex) `GET baseURI/job/750x00000000014/batch/751x00000000030/result/752x000000000`

### CODE REVIEW

File: src/main/java/com/springml/salesforce/wave/api/BulkAPI.java [hunk 1]
```diff
@@ -88,4 +89,23 @@
...
     public BatchInfo getBatchInfo(String jobId, String batchId) throws Exception;
+
+    /**
+     * Retrieve a list of batch result IDs for a particular batch job.
+     * @param jobId The identifier for the job
+     * @param batchId The batch identifier for the job
+     * @return
+     * @throws Exception
+     */
+    public List<String> getBatchResultIds(String jobId, String batchId) throws Exception;
```
[the remark this answers is not in the document]
Comment (samuel-pt) [reply src/main/java/com/springml/salesforce/wave/api/BulkAPI.java]: @mengxr No problem with adding a method as spark-salesforce is the only downstream project

**Comment by k4jiang** [comment]
Do not merge this PR yet. The API for fetching the Batch Result IDs from Salesforce is not working at the moment. It is returning two very different responses. Example of the two different formats:
1) `[123312321,13214214]`
2) `<result-list xmlns=\"http://www.force.com/2009/06/asyncapi/dataload\"><result>123312321,13214214</result><result>123312321,13214214</result></result-list>`

**Comment by samuel-pt** [comment]
As mentioned earlier, please prefix the title it with [NO MERGE]

**Comment by k4jiang** [comment]
The issue is fixed. Feel free to merge whenever ready.

**Pull request merged**